app/src/main/java/com/github/se/cyrcle/ui/addParking/location/overlay/RectangleSelection.kt

```diff
@@ -56,15 +65,26 @@ fun RectangleSelection(mapViewModel: MapViewModel, paddingValues: PaddingValues)
           }
           width = touchPosition.x - canvasCenter.x
           height = touchPosition.y - canvasCenter.y
-          // Convert width o the same unit as the canvas center
+          // Convert width to the same unit as the canvas center
           val rectSize = Size(width, height)
+
+          val listScreenCoordinates = computeScreenCoordinates(canvasCenter, width, height)
+          val pointsList = mapView?.mapboxMap?.coordinatesForPixels(listScreenCoordinates)
+          val area = pointsList?.let { Location(it).computeArea() } ?: 0.0
+          mapViewModel.updateIsAreaTooLarge(area > PARKING_MAX_AREA)
```

Comment:
You got me thinking that maybe it would be better to not have area in the viewModel and just use it in the screen? What do you think?

Reply:
I think it's fine to keep isAreaValid in the viewmodel, since you reuse it multiple components. 
We do it all the time in the project, it's the most readable way to do it in my opinion